helm/cluster-openstack/templates/_machine_deployment.tpl

```diff
@@ -0,0 +1,31 @@
+{{- define "machine-deployment" }}
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  clusterName: {{ include "resource.default.name" $ }}
+  replicas: {{ .replicas }}
+  selector:
+    matchLabels: null
+  template:
+    metadata:
+      labels:
+        {{- include "labels.common" $ | nindent 8 }}
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+          name: {{ include "resource.default.name" $ }}-{{ .class }}
```

Comment:
[here](https://github.com/giantswarm/cluster-openstack/pull/71/files#diff-d9842967017d1ccef771e785c633a6b8b429882234430dad2e6907b27d692a1aL34) we used `name` instead of `class`. Same for `infrastructureRef` part

Reply:
We generate KubeAdmConfigTemplate and OpenStackMachineTemplate per nodeclass here https://github.com/giantswarm/cluster-openstack/blob/main/helm/cluster-openstack/templates/list.yaml#L11 

We generate MachineDeployment per nodePool and every nodepool gives a reference to a nodeclass. That is why we refer to class here.